src/renderer.tsx

```diff
@@ -20,18 +33,60 @@ export const NotionRenderer: React.FC<NotionRendererProps> = ({
   const id = currentId || Object.keys(blockMap)[0];
   const currentBlock = blockMap[id];
 
-  if (!currentBlock) return null;
+  if (!currentBlock) {
+    if (process.env.NODE_ENV !== "production") {
+      console.warn("error rendering block", currentId);
+    }
+    return null;
+  }
+
+  const mapPageUrl = props.mapPageUrl || defaultMapPageUrl(props.rootPageId);
+  const zoom =
+    props.zoom ||
+    (typeof window !== "undefined" &&
+      mediumZoom({
+        container: ".notion",
+        background: "rgba(0, 0, 0, 0.8)",
+        margin: getMediumZoomMargin()
+      }));
 
   return (
-    <Block key={id} level={level} block={currentBlock} {...props}>
+    <Block
+      key={id}
+      level={level}
+      block={currentBlock}
+      zoom={zoom}
+      mapPageUrl={mapPageUrl}
+      {...props}
+    >
       {currentBlock?.value?.content?.map(contentId => (
         <NotionRenderer
           key={contentId}
           currentId={contentId}
           level={level + 1}
+          zoom={zoom}
+          mapPageUrl={mapPageUrl}
           {...props}
         />
       ))}
     </Block>
   );
 };
+
+function getMediumZoomMargin() {
```

Comment:
this is my lame attempt to roughly match notion's behavior for the size of the zoomed image lightbox